### PR TITLE
IGNITE-19668 Sql. Fix testLengthExpressionWithDynamicParameter

### PR DESCRIPTION
**Pull request by zstan**

### CODE REVIEW

**Comment by zstan** [comment]
@lowka can u make a review plz ?

**Review by lowka — APPROVED** [review]

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/datatypes/varbinary/ItVarBinaryExpressionTest.java [hunk 1]
```diff
@@ -67,24 +67,32 @@ public void testPositionExpressionWithDynamicParameter() {
...
+        checkQuery("SELECT OCTET_LENGTH(?)")
+                .withParams(varBinary(new byte[0]))
+                .returns(0).check();
+
+        checkQuery("SELECT LENGTH(CAST (? AS VARCHAR))")
                 .withParams(varBinary(new byte[]{1, 2, 3}))
                 .returns(3).check();
 
-        checkQuery("SELECT LENGTH(?)")
+        checkQuery("SELECT LENGTH(CAST (? AS VARCHAR))")
```
Comment (korlov42) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/datatypes/varbinary/ItVarBinaryExpressionTest.java:95]: why do you put CAST operator for last two cases?

Comment (zstan) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/datatypes/varbinary/ItVarBinaryExpressionTest.java:95]: just to make sure that it works with cast too

Comment (korlov42) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/datatypes/varbinary/ItVarBinaryExpressionTest.java:95]: by "it" did you mean LENGTH?

Comment (zstan) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/datatypes/varbinary/ItVarBinaryExpressionTest.java:95]: yes, cast from varbinary to varchar returns the expected result.

Comment (korlov42) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/datatypes/varbinary/ItVarBinaryExpressionTest.java:95]: well, and what is expected result in this case?

**Comment by zstan** [comment]
@lowka @korlov42 new approach is ready for review, can u take a look plz ?

**Review by lowka — APPROVED** [review]

**Pull request merged**